### DISCUSSION TITLE
Vanilla extract equivalent of stitches responsive variants

### DISCUSSION THREAD

**Discussion by Code-Victor** [vanilla-extract-css/vanilla-extract#1125] [Q&A]
I'm coming from [stitches](https://stitches.dev/) CSS-in-JS and I'm looking for a similar API so I can migrate easily. So far, I've found [Recipes](https://vanilla-extract.style/documentation/packages/recipes/) and [Sprinkles](https://vanilla-extract.style/documentation/packages/sprinkles/) which solve all my issues except for the [responsive variants](https://stitches.dev/docs/responsive-styles#responsive-variants). Is there any way to have a similar feature in vanilla extract?
Thank you.

**Comment by lloydjatkinson** [comment]
Also interested!

**Reply by vatoutou** [reply]
Did you try to use sprinkles conditions? 👀 
In my sprinkles I have:
```ts
conditions: {
	base: {}, /* Mobile. */
	desktop: { '@media': breakpoints.desktop },
	tablet: { '@media': breakpoints.tablet },
	hover: { selector: '&:hover' },
	active: { selector: '&:active' },
},
```
So  I can use it like:

```ts
sprinkles({
	color: {
		base: 'red',
        	tablet: 'white',
		desktop: 'violet',
	}
})
```

**Reply by lloydjatkinson** [reply]
Do you have an example of using that in components?

**Reply by 6mint** [reply]
You can use the sprinkles function in .css.ts as well as in components. So a react example would be:
```tsx
<div
  className={sprinkles({
    color: {
      base: "red",
      tablet: "white",
      desktop: "violet",
    },
  })}
/>
```

**Reply by 6mint** [reply]
A corresponding sprinkles definition could look like this:
```ts
const colorProperties = defineProperties({
  conditions: {
    base: {} /* Mobile. */,
    tablet: { "@media": "@media (width >= 768px)" },
    desktop: { "@media": "@media (width >= 1024px)" },
  },
  defaultCondition: "base",
  properties: {
    color: {
        red: "#ff0000",
        white: "#ffffff",
        violet: "#571671",
    },
  },
});

export const sprinkles = createSprinkles(
    colorProperties
);
```
